content/blog/scale-kubernetes-clusters-using-hpe-greenlake-terraform-provider.md

```diff
@@ -0,0 +1,399 @@
+---
+IaC or Infrastructure as code is a practice of automating the process of managing and provisioning infrastructure through code instead of through manual processes. It gives organizations the tools to create, manage, and destroy compute resources by statically defining and declaring these resources in codeblocks. It helps to increase operational agility, simplify management, reduce errors, and save cost.
+
+I﻿n this post, I will explore options to declare and upgrade Kubernetes clusters on HPE GreenLake using the HPE GreenLake Terraform Provider.
+
+A﻿s the part of upgrade, the following 3 scenarios are supported --  
+
+1. Scaling of a cluster's worker nodes. Please refer to the blog post [Scale Kubernetes Clusters using HPE GreenLake Terraform Provider](https://developer.hpe.com/blog/scale-kubernetes-cluster-using-hpe-greenlake-terraform-provider/) to check out scaling options available for worker nodes.
+2. Upgrade the Kubernetes version of the cluster. This step is covered in this blog.
+3. Upgrade the OS version of the worker nodes inside a cluster. This step is covered in this blog.
+
+# Prerequisite
+
+Before starting this tutorial, read the blog post [Kubernetes Cluster as Code - Part 1](https://developer.hpe.com/blog/kubernetes-clusters-as-code-part1/), which includes steps for creating a Kubernetes cluster. This post expands upon that scenario by examining how to upgrade a cluster's Kubernetes version & OS version of its worker nodes.
+
+# Verify existing Kubernetes cluster
+
+After the cluster is created following the instructions found in the  [Kubernetes Cluster as Code - Part 1 blog post](https://developer.hpe.com/blog/kubernetes-clusters-as-code-part1/), launch HPE GreenLake Central console and verify that the cluster is present under the appropriate tenant.
+
+You should see the  **tf-test**  cluster present under  **Dashboard -> Manage your Private Cloud -> Containers**.
+
+![](https://developer.hpe.com/img/cluster_list.jpg)
+
+Below is the reference Terraform configuration file for the existing cluster.
+
+```hcl
+terraform {
+  required_providers {
+    hpegl = {
+      source  = "hpe/hpegl"
+      version = ">= 0.2.2"
+    }
+  }
+}
+ 
+provider hpegl {
+  caas {
+  }
+}
+ 
+variable "HPEGL_SPACE" {
+  type = string
+}
+ 
+data "hpegl_caas_site" "blr" {
+  name = "BLR"
+  space_id = var.HPEGL_SPACE
+ }
+ 
+data "hpegl_caas_cluster_blueprint" "bp" {
+  name = "demo"
+  site_id = data.hpegl_caas_site.blr.id
+}
+ 
+resource hpegl_caas_cluster test {
+  name         = "tf-test"
+  blueprint_id = data.hpegl_caas_cluster_blueprint.bp.id
+  site_id      = data.hpegl_caas_site.blr.id
+  space_id     = var.HPEGL_SPACE
+}
+```
+
+# Upgrade the Kubernetes version
+
+For the Kubernetes version upgrade, you need to specify the new version of Kubernetes that is available for upgrade in the resources block. 
+
+1. **kubernetes_version**: *Use the Kubernetes version that pops up on the cluster details page in the UI.*
```

Comment:
Is it possible to get that information programmatically? 
if yes, it would be interesting to mention it. Even if for this blog on terraform, the easy way is from the UI.

Reply:
No its not possible to get it programmatically.